chore: Update to .NET 10 Preview 6

jonpryor opened the pull request:
Context: https://github.com/unoplatform/uno/pull/19934
Context: https://github.com/dotnet/sdk/issues/48949
Context: https://github.com/dotnet/android/commit/70bd636b04ebf9dba36c632cdec586452e98cb8a

unoplatform/uno#19934 was the first attempt to build unoplatform/uno with .NET 10.  It currently *builds* (mostly), but it has *lots* of unit test failures (over 3700).

Reviewing the diff between master@8e93e0d5 and
dev/jela/ref-updates@81c222f90c9704233ed2a8e0e8ee258dbda4c8e6 shows lots of changes which are apparently unrelated to getting Uno building atop .NET 10.  Could some of those changes be responsible?

Test out this hypothesis by applying the .NET 10 changes to a *new* branch + PR, which will allow us to see determine if *just* building atop .NET 10 is the cause of the unit test failures.

To be clear, ".NET 10 changes" still constitutes *a lot* of changes. Broadly speaking, they fall under:

  * `$(TreatWarningsAsErrors)`=true + NuGet warnings.  For example:

        error NU1510: Warning As Error:
        PackageReference System.Private.Uri will not be pruned.
        Consider removing this package from your dependencies, as it is likely unnecessary.

  * `$(TreatWarningsAsErrors)`=true + IDE0055 warnings, which mostly involved adding blank lines between `namespace …` declarations and type declarations.

  * `$(TreatWarningsAsErrors)`=true + CS8604 warnings due to nullability changes, particularly in iOS.

  * `$(TreatWarningsAsErrors)`=true + IL2\* warnings, some of which I was able to fix in a reasonable timeframe, and some of which I just threw into `$(NoWarn)` to deal with later.

  * Workarounds for dotnet/sdk#48949, specifically altering `%(KnownFrameworkReference.*Version)` metadata so that the correct version of `Microsoft.WindowsDesktop.App.Ref` is found.

  * Workarounds for dotnet/android@70bd636b, which introduced a new `Microsoft.Android` namespace into `Mono.Android.dll`, which causes a CS0234 error when inside a `Microsoft.` namespace and C# code tries to use e.g. `Android.Views.View`.

  * Use .NET 9 to run `dotnet tool update` on Linux, to avoid:

        Unhandled exception: Command workload has more than one child named "--info"

    [This will hopefully be fixed in .NET 10 Preview 7][0].

[0]: https://discord.com/channels/732297728826277939/1267023991428612199/1397307201919652023

**GitHub Issue:** closes #

<!-- Link to relevant GitHub issue if applicable. All PRs should be associated with an issue (GitHub issue or internal), unless the change is documentation related. -->

## PR Type:

<!--
Copy the labels that apply to this PR and paste them above:

- 🐞 Bugfix
- ✨ Feature
- 🎨 Code style update (formatting)
- 🔄 Refactoring (no functional changes, no api changes)
- 🏗️ Build or CI related changes
- 📚 Documentation content changes
- 🤖 Project automation
- 💬 Other... (Please describe)

-->


## What is the current behavior? 🤔

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->


## What is the new behavior? 🚀

<!-- Please describe the new behavior after your modifications. -->

## PR Checklist ✅

Please check if your PR fulfills the following requirements:

- [ ] 📝 Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.
- [ ] 🧪 Added [Runtime tests, UI tests, or a manual test sample](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] 📚 Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] 🖼️ Validated PR `Screenshots Compare Test Run` results.
- [ ] ❗ Contains **NO** breaking changes

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below.
     Please note that breaking changes are likely to be rejected -->

## Other information ℹ️

<!-- Please provide any additional information if necessary -->

jonpryor committed:
chore: Update to .NET 10 Preview 6

Context: https://github.com/unoplatform/uno/pull/19934
Context: https://github.com/dotnet/sdk/issues/48949
Context: https://github.com/dotnet/android/commit/70bd636b04ebf9dba36c632cdec586452e98cb8a

unoplatform/uno#19934 was the first attempt to build unoplatform/uno
with .NET 10.  It currently *builds* (mostly), but it has *lots* of
unit test failures (over 3700).

Reviewing the diff between master@8e93e0d5 and
dev/jela/ref-updates@81c222f90c9704233ed2a8e0e8ee258dbda4c8e6
shows lots of changes which are apparently unrelated to getting Uno
building atop .NET 10.  Could some of those changes be responsible?

Test out this hypothesis by applying the .NET 10 changes to a
*new* branch + PR, which will allow us to see determine if *just*
building atop .NET 10 is the cause of the unit test failures.

To be clear, ".NET 10 changes" still constitutes *a lot* of changes.
Broadly speaking, they fall under:

  * `$(TreatWarningsAsErrors)`=true +
    NuGet warnings.  For example:

        error NU1510: Warning As Error:
        PackageReference System.Private.Uri will not be pruned.
        Consider removing this package from your dependencies, as it is likely unnecessary.

  * `$(TreatWarningsAsErrors)`=true + IDE0055 warnings, which mostly
    involved adding blank lines between `namespace …` declarations
    and type declarations.

  * `$(TreatWarningsAsErrors)`=true + CS8604 warnings due to
    nullability changes, particularly in iOS.

  * `$(TreatWarningsAsErrors)`=true + IL2\* warnings, some of which I
    was able to fix in a reasonable timeframe, and some of which I
    just threw into `$(NoWarn)` to deal with later.

  * Workarounds for dotnet/sdk#48949, specifically altering
    `%(KnownFrameworkReference.*Version)` metadata so that the
    correct version of `Microsoft.WindowsDesktop.App.Ref` is found.

  * Workarounds for dotnet/android@70bd636b, which introduced a new
    `Microsoft.Android` namespace into `Mono.Android.dll`, which causes
    a CS0234 error when inside a `Microsoft.` namespace and C# code
    tries to use e.g. `Android.Views.View`.

  * Use .NET 9 to run `dotnet tool update` on Linux, to avoid:

        Unhandled exception: Command workload has more than one child named "--info"

    [This will hopefully be fixed in .NET 10 Preview 7][0].

[0]: https://discord.com/channels/732297728826277939/1267023991428612199/1397307201919652023

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21140/wasm-skia-net9/index.html

jonpryor committed:
chore: Fix Tests - Unit Tests build

**Tests - Unit Tests > Run > Build unit tests solution filter** was
failing with:

	C:\a\1\s\src\Uno.UI.RemoteControl.TestProcessor\Uno.UI.RemoteControl.TestProcessor.csproj :
	error NU1201: Project Uno.UI.RemoteControl.Server is not compatible with net8.0 (.NETCoreApp,Version=v8.0).
	  Project Uno.UI.RemoteControl.Server supports:
	    - net10.0 (.NETCoreApp,Version=v10.0)
	    - net9.0 (.NETCoreApp,Version=v9.0)

This is because `Uno.UI.RemoteControl.TestProcessor.csproj` was
explicitly targeting `net8.0`, while its dependencies were using
`$(NetPrevious);$(NetCurrent)`.

Update `Uno.UI.RemoteControl.TestProcessor.csproj` to use
`$(NetPrevious)`, so that we don't need to worry about it in the future.

Fix new IDE0055 warnings.

Review `git grep net8.0` output.  Ignoring `.nuspec` files and
`src/SolutionTemplate`, the only remaining use of `net8.0` is in
`src/Uno.UWPSyncGenerator/Generator.cs`, a'la:

	_androidCompilation = await LoadProject($@"{topProject}.netcoremobile.csproj", "net8.0-android");

Update these to `net9.0`, just in case.

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21140/wasm-skia-net9/index.html

unodevops commented:
⚠️⚠️ The build [171690](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=171690) has failed on Uno.UI - CI.

jonpryor committed:
chore: Fix Tests - Templates

**Tests - Templates > macOS Tests group_1 > Run Project Template Tests**
was failing with:

	error NETSDK1147: To build this project, the following workloads must be installed: android-net8
	error NETSDK1147: To install these workloads, run the following command: dotnet workload restore

Uno only supports two .NET versions, `$(NetCurrent)` and
`$(NetPrevious)`.  With the addition of .NET 10, that means that .NET 8
falls out of support, and thus we should not be building these projects
against .NET 8.

Some of the `src/SolutionTemplate` projects already made use of a
`$(DotNetVersion)` property.

Add new `$(DotNetCurrentVersion)` and `$(DotNetPreviousVersion)`
MSBuild properties to  `src/SolutionTemplate/Directory.Build.props`
so that .NET versions can be used consistently, with
`$(DotNetPreviousVersion)`=net9.0, `$(DotNetCurrentVersion)`=net10.0.
This allows nearly everything else to be based upon these new
properties, e.g. `net8.0-android` becomes
`$(DotNetPreviousVersion)-android`, which should make it easier to
bump this value in the future.

Update use of `net8.0-android34.0` to `net9.0-android35.0`.

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21140/wasm-skia-net9/index.html

jonpryor committed:
chore: fix local-android-uitest-run.sh et al to work on macOS

Export and use `$ANDROID_AVD_HOME`, so that we have a "known" location
for whare Emulator images are created.  (Not sure what's wrong with
my environment, but they are not reliably created or loaded from
`$HOME/.android/avd`.)

Update `local-android-uitest-run.sh` so that it doesn't require running
the script from the `build/test-scripts` directory.  You can now run it
from topdir:

	UITEST_IS_LOCAL=true \
	  build/test-scripts/local-android-uitest-run.sh

Update the expected paths to match .NET 9 output.

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21140/wasm-skia-net9/index.html

unodevops commented:
⚠️⚠️ The build [171806](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=171806) has failed on Uno.UI - CI.

jonpryor committed:
chore: *don't* use the dotnet-install.yml template?

Context: 5c306767f7f0a8e0ed056ed81188658bf2d9d6e0
Context: https://github.com/unoplatform/uno/pull/19934

The **Tests - Desktop Skia Windows > … > Using .NET 10.0 global.json**
and **Tests - Desktop Skia Linux > … > Using .NET 10 global.json**
steps are each failing, for the same reason:

	rm: cannot remove 'global.json': No such file or directory
	cp: cannot stat 'build/ci/net10/global.json': No such file or directory

The reason why this step is failing is twofold:

 1. Each of those jobs have `checkout: none`, i.e. they do not check
    out the `unoplatform/uno` repo, and

 2. They each run the `dotnet-install.yml` template, which in turn is
    trying to access files from the (nonexistent!) checkout.

This change was introduced in commit 5c306767, which in turn was
"cribbed" from unoplatform/uno#19934.

Is the inclusion of `dotnet-install.yml` necessary?

Remove it, and let's find out!

jonpryor committed:
chore: fix Uno.UI.RuntimeTests.HRApp.Skia.csproj

The **Tests - Desktop Skia Windows > Runtime Tests** step fails:

	HRAppBuild: D:\a\1\s\src\Uno.UI.RuntimeTests\Tests\HotReload\Frame\HRApp\Uno.UI.RuntimeTests.HRApp.Skia.csproj :
	error NU1015: The following PackageReference item(s) do not have a version specified: Uno.Fonts.OpenSans
	HRAppBuild: D:\a\1\s\src\Uno.UI.RuntimeTests\Tests\HotReload\Frame\HRApp\Uno.UI.RuntimeTests.HRApp.Skia.csproj :
	warning NU1510: PackageReference Microsoft.Win32.Registry will not be pruned. Consider removing this package from your dependencies, as it is likely unnecessary.

Address these messages:

  * Add `%(Version)` metadata for the `Uno.Fonts.OpenSans` package.
  * Remove the `@(PackageReference)` for `Microsoft.Win32.Registry`.

unodevops commented:
⚠️⚠️ The build [171824](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=171824) has failed on Uno.UI - CI.

jonpryor committed:
chore: selectively copy global.json in dotnet-install.yml

Context: d20e596c74f1638e77a80ccafe2041512e209f87

In commit d20e596c we attempted to fix e.g.
**Tests - Desktop Skia Linux** by *removing* use of
`dotnet-install.yml`, asking:

> Is the inclusion of `dotnet-install.yml` necessary?

Yes, yes it is.  On the current run,
**Tests - Desktop Skia Lilnux > Snapshot Tests > Run Snapshot Tests**
fails with:

	You must install or update .NET to run this application.

	App: /home/vsts/work/1/s/build/skia-generic-samples-app-WinUI/SamplesApp.Skia.Generic.dll
	Architecture: x64
	Framework: 'Microsoft.NETCore.App', version '10.0.0-preview.6.25358.103' (x64)
	.NET location: /usr/share/dotnet/

	The following frameworks were found:
	  6.0.36 at [/usr/share/dotnet/shared/Microsoft.NETCore.App]
	  8.0.18 at [/usr/share/dotnet/shared/Microsoft.NETCore.App]
	  9.0.4 at [/usr/share/dotnet/shared/Microsoft.NETCore.App]

What's happening is the "Downloading skia-generic-samples-app-WinUI"
step downloads binaries built from a previous stage, and those binaries
were built against -- and thus *require* -- .NET 10.

Meanwhile, the default bot does not have .NET 10.  This is why the
`dotnet-install.yml` script was required.

However, as per d20e596c, we *know* that using the `dotnet-install.yml`
script will fail, as-is, because there is no checokut.

Update `dotnet-install.yml` to accept an optional `haveCheckout`
parameter (default `true`), and have
`azure-devops-tests-*-skia.yml` provide `haveCheckout: false`.
Hopefully this way we can *skip* the attempt to copy the nonexistent
`global.json` file, while still installing the required .NET version.

Further update `dotnet-install.yml` so that `UseDotNet@2` doesn't
always require `global.json`, and will install
`defaultDotNetSdkVersion` if no `global.json` is present, i.e.
when `haveCheckout: false`.

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21140/wasm-skia-net9/index.html

unodevops commented:
⚠️⚠️ The build [171892](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=171892) has failed on Uno.UI - CI.

jonpryor commented:
**Tests - WebAssembly Skia > Run Runtime Tests \* > Running UI Tests** is failing due to: https://github.com/unoplatform/Uno.Wasm.Bootstrap/issues/986

jonpryor committed:
Merge remote-tracking branch 'origin/master' into dev/jonp/jonp-build-with-net10

jonpryor committed:
chore: Provision Microsoft OpenJDK-17

Context: https://github.com/dotnet/android/commit/df68c208ad02d09ebb9e6a87053343985dfc1e36

.NET for Android in .NET 9 bumped the minimum supported JDK version
from JDK-11 to JDK-17.  Now that we're bumping the supported .NET
versions from (net8.0, net9.0) to (net9.0, net10.0), and thus many
of the template tests are now building against .NET 9.0, some tests
are failing; consider **Tests - Templates > macOS Tests group_4**:

	MANIFESTMERGER : error : LinkageError occurred while loading main class com.xamarin.manifestmerger.Main
	MSBUILD : java error AMM0000: Error: LinkageError occurred while loading main class com.xamarin.manifestmerger.Main
	MSBUILD : java error AMM0000: java.lang.UnsupportedClassVersionError
	MSBUILD : java error AMM0000:  com/xamarin/manifestmerger/Main has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 55.0

Fix this by using JDK-17 on the bots instead of JDK-11.

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21140/wasm-skia-net9/index.html

jonpryor committed:
chore: fix Build - Managed Binaries

Context: b0f158128ecbcdc68f278b4521a1f4e8e155dfa1

Merge commit b0f15812 broke the **Build - Managed Binaries** stage
as it inadvertently removed the inclusion of `GlobalUsings.cs` from
`Uno.UI.Runtime.Skia.Android.csproj`.

Add that back.

jonpryor committed:
chore: Use Xcode 16.4

The **Tests - Templates > macOS Tests group_3** job is failing
because .NET for iOS requires Xcode 16.4:

	Xamarin.Shared.Sdk.targets(2339,3): error : This version of .NET for iOS (18.5.10415-net10-p6) requires Xcode 16.4.
	  The current version of Xcode is 16.2.
	  Either install Xcode 16.4, or use a different version of .NET for iOS. See https://aka.ms/xcode-requirement for more information.

Hopefully Xcode 16.4 is installed on the macOS bots…

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21140/wasm-skia-net9/index.html

unodevops commented:
⚠️⚠️ The build [172131](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=172131) has failed on Uno.UI - CI.

jonpryor committed:
chore: "fix" Tests - Templates > * Tests group_2

Context: https://github.com/unoplatform/uno/commit/9cbb2297070#diff-15b07bd6990cd29f75427f6289eda5c4c0958ac89ba0b1a34575cc5b2419b574

**Tests - Templates > macOS Tests group_2 > Run Project Templates Tests**
fails with:

	/Users/runner/work/1/s/src/SolutionTemplate/5.2/uno52AppWithLib/uno52AppWithLib/uno52AppWithLib.csproj(101,5):
	error : Missing asset package_b3a9c656b45bb0f580b63c42f90f90863a84f337/Assets/SharedAssets.md

The `src/SolutionTemplate/5.2/uno52AppWithLib` project appears to be
similar, structurally, to `src/SolutionTemplate/5.3/uno53AppWithLib`,
with uno52AppWithLib originally targeting .NET 8, while
uno53AppWithLib originally targeted .NET 9.  In the process, the
.NET 9-based uno53AppWithLib app does *not* assert that `test.xml`
is in `$(OutputPath)`, with a comment stating:

	Transitive library asset are not included in the output folder, as they're referenced
	in-place by `uno53AppWithLib.staticwebassets.runtime.json`

Assuming that this is still the case, remove `test.xml` and
`icon_lib.png` from `@(_AssetsToValidate)`.  This allows the project
to build successfully.

TODO: is `src/SolutionTemplate/5.2` needed anymore?  Should we just
remove it from the tests?

jonpryor committed:
chore: (Attempt to) Fix Tests - Templates > Linux Tests*

The **Tests - Templates > Linux Tests group_\*** jobs are all failing
for various reasons.

**Linux Tests group_0** is failing with:

	/agent/_work/1/s/src/SolutionTemplate/UnoAppWinUILinuxValidation/UnoAppWinUILinuxValidation.Base/AppHead.xaml.cs(41,17):
	error CS0012: The type 'ILoggerFactory' is defined in an assembly that is not referenced. You must add a reference to assembly 'Microsoft.Extensions.Logging.Abstractions, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'.
	[/agent/_work/1/s/src/SolutionTemplate/UnoAppWinUILinuxValidation/UnoAppWinUILinuxValidation.Wasm/UnoAppWinUILinuxValidation.Wasm.csproj]

This project is currently referencing .NET *6* packages, which is
ancient.

Update `run-net7-template-linux.ps1` to stop running tests for
Uno 5.0 and 5.1.

TODO: Revisit this.

**Linux Tests group_1** is failing with:

	…/Microsoft.NET.Sdk.StaticWebAssets/targets/Microsoft.NET.Sdk.StaticWebAssets.Compression.targets(250,5): error MSB4018: The "ApplyCompressionNegotiation" task failed unexpectedly. [/agent/_work/1/s/src/SolutionTemplate/5.1/uno51recommended/uno51recommended.Wasm/uno51recommended.Wasm.csproj]
	…/Microsoft.NET.Sdk.StaticWebAssets/targets/Microsoft.NET.Sdk.StaticWebAssets.Compression.targets(250,5): error MSB4018: System.InvalidOperationException: Endpoints not found for related asset: /agent/_work/1/s/src/SolutionTemplate/5.1/uno51recommended/uno51recommended.Wasm/bin/Debug/net9.0/dist/favicon.ico [/agent/_work/1/s/src/SolutionTemplate/5.1/uno51recommended/uno51recommended.Wasm/uno51recommended.Wasm.csproj]
	…/Microsoft.NET.Sdk.StaticWebAssets/targets/Microsoft.NET.Sdk.StaticWebAssets.Compression.targets(250,5): error MSB4018:    at Microsoft.AspNetCore.StaticWebAssets.Tasks.ApplyCompressionNegotiation.ResolveEndpoints(IDictionary`2 assetsById, IDictionary`2 endpointsByAsset, StaticWebAsset compressedAsset) [/agent/_work/1/s/src/SolutionTemplate/5.1/uno51recommended/uno51recommended.Wasm/uno51recommended.Wasm.csproj]
	…/Microsoft.NET.Sdk.StaticWebAssets/targets/Microsoft.NET.Sdk.StaticWebAssets.Compression.targets(250,5): error MSB4018:    at Microsoft.AspNetCore.StaticWebAssets.Tasks.ApplyCompressionNegotiation.Execute() [/agent/_work/1/s/src/SolutionTemplate/5.1/uno51recommended/uno51recommended.Wasm/uno51recommended.Wasm.csproj]
	…/Microsoft.NET.Sdk.StaticWebAssets/targets/Microsoft.NET.Sdk.StaticWebAssets.Compression.targets(250,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(TaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/agent/_work/1/s/src/SolutionTemplate/5.1/uno51recommended/uno51recommended.Wasm/uno51recommended.Wasm.csproj]

Removing the Uno 5.1 tests will also fix (hide) this error.

**Linux Tests group_2** should be fixed with f9f579a2.

**Linux Tests group_3** fails with:

	…/uno53net9blank.csproj : error NU1102: Unable to find package Microsoft.WindowsDesktop.App.Ref with version (= 10.0.0-preview.6.25315.102)
	…/uno53net9blank.csproj : error NU1102:   - Found 146 version(s) in NuGet official package source [ Nearest version: 10.0.0-preview.6.25358.103 ]
	…/uno53net9blank.csproj : error NU1102:   - Found 0 version(s) in /agent/_work/1/s/.dotnet/library-packs
	…/uno53net9blank.csproj : error NU1102:   - Found 0 version(s) in Solution Packages
	…/uno53net9blank.csproj : error NU1102:   - Found 0 version(s) in uno-dev

This is the same scenario that 5c306767 mentioned:

> * Workarounds for dotnet/sdk#48949, specifically altering
>   `%(KnownFrameworkReference.*Version)` metadata so that the
>   correct version of `Microsoft.WindowsDesktop.App.Ref` is found.

Apply the same workaround.

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21140/wasm-skia-net9/index.html

jonpryor committed:
Merge remote-tracking branch 'origin/master' into dev/jonp/jonp-build-with-net10

jonpryor committed:
chore: Uno.WinUI.Graphics2DSK.nuspec targets .NET 9+10

Context: 69f1d49902dbdf0a04cb14a6211d3b0120ddcc9e
Context: 5c8d0e8b142c2c5e27d6c1a8eb3d61a504a801f6

I push 69f1d499, expecting to see progress on **Tests - Templates**.
Instead, **Build - Managed Binaries > Build > Generate NuGet Packages**
fails:

	EXEC : warning : NU5130: Some target frameworks declared in the dependencies group of the nuspec and the lib/ref folder have compatible matches, but not exact matches in the other location. Unless intentional, consult the list of actions below: [C:\a\1\s\Build\Uno.UI.Build.csproj]
	  - Add a dependency group for .NETFramework9.0-windows10.0.19041.0 to the nuspec
	  NuGet Version: 5.4.0.6315
	  Attempting to build package from 'Uno.WinUI.Graphics2DSK.nuspec'.
	  Could not find a part of the path 'C:\a\1\s\src\AddIns\Uno.WinUI.Graphics2DSK\bin\Release\net8.0'.
	  System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\a\1\s\src\AddIns\Uno.WinUI.Graphics2DSK\bin\Release\net8.0'.
	     at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
	     …
	C:\a\1\s\Build\Uno.UI.Build.csproj(355,3): error MSB3073: The command ".\external\nuget\NuGet.exe pack nuget\Uno.WinUI.Graphics2DSK.nuspec -Verbosity Detailed -Version "6.2.0-dev.154.g2b2e4d1b79" -Properties "NoWarn=NU5100,NU5105,NU5131;NU5123;branch=2b2e4d1b791cd1ed229b5bdf08ff32c113990042;commitid=2b2e4d1b791cd1ed229b5bdf08ff32c113990042;winuisourcepath=net9.0-windows10.0.19041.0;winuitargetpath=net9.0-windows10.0.19041.0"" exited with code 1.

It occurs to me that what probably happened is origin/master was
implicitly merged.  There were no merge conflicts, but this brought
in 5c8d0e8b, which reintroduces references to .NET 8 artifacts.

Merge in origin/master (bc5cc040), then update
`build/nuget/Uno.WinUI.Graphics2DSK.nuspec` so that .NET 8 artifacts
are no longer referenced.

This *should* fix the **Generate NuGet Packages** step.

jonpryor committed:
chore: Uno.WinUI.Graphics2DSK.nuspec only targets .NET 9 Windows

Context: b789db7f0a58cd900dffc831a75e94cec8ef3b52

Commit b789db7f tried to fix **Build - Managed Binaries**, but failed:

	  Attempting to build package from 'Uno.WinUI.Graphics2DSK.nuspec'.
	  Could not find a part of the path 'C:\a\1\s\src\AddIns\Uno.WinUI.Graphics2DSK\bin\Release\net10.0-windows10.0.19041.0'.
	  System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\a\1\s\src\AddIns\Uno.WinUI.Graphics2DSK\bin\Release\net10.0-windows10.0.19041.0'.
	     at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
	     …
	C:\a\1\s\Build\Uno.UI.Build.csproj(355,3): error MSB3073: The command ".\external\nuget\NuGet.exe pack nuget\Uno.WinUI.Graphics2DSK.nuspec -Verbosity Detailed -Version "6.2.0-dev.156.g094e6f1fa9" -Properties "NoWarn=NU5100,NU5105,NU5131;NU5123;branch=094e6f1fa9dc5606d8aea64265ec1800fd1fef4d;commitid=094e6f1fa9dc5606d8aea64265ec1800fd1fef4d;winuisourcepath=net9.0-windows10.0.19041.0;winuitargetpath=net9.0-windows10.0.19041.0"" exited with code 1.

Reviewing `src/AddIns/Uno.WinUI.Graphics2DSK/Uno.WinUI.Graphics2DSK.Windows.csproj`,
and it *only* builds for `$(NetPreviousWinAppSDK)`, i.e. .NET 9 now.

Update `build/nuget/Uno.WinUI.Graphics2DSK.nuspec` so that it doesn't
attempt to package net10.0-windows10\* binaries.

unodevops commented:
⚠️⚠️ The build [172373](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=172373) has failed on Uno.UI - CI.
